pkg/services/store/entity/health.go

```diff
@@ -0,0 +1,68 @@
+package entity
+
+import (
+	"context"
+
+	grpcAuth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+// Compile-time assertion
+var _ HealthService = &healthServer{}
+
+type HealthService interface {
+	grpc_health_v1.HealthServer
+	grpcAuth.ServiceAuthFuncOverride
+}
+
+type InitializableEntityServer interface {
+	EntityStoreServer
+	Init() error
+}
+
```

Comment:
It may make sense to just extend the EntityStoreServer interface to expose `IsHealthy() bool`